src/mui/input/SelectArrayInput.js

```diff
@@ -130,9 +178,44 @@ export class SelectArrayInput extends Component {
         };
     }
 
+    renderCreateButton = () => (
+        <MenuItem
+            rightIcon={<AddIcon color={cyan500} />}
+        >
+            <span
+                style={{
+                    fontWeight: 'bold',
+                    overflow: 'hidden',
+                    textOverflow: 'ellipsis',
+                    display: 'block',
+                }}
+            >
+                Create: {this.state.currentValue}
+            </span>
+        </MenuItem>
+    );
+
+    getChoices = () => {
+        const exists = this.existsInChoices(this.state.currentValue);
+        const showCreateButton = this.props.onCreateInline && !exists && this.state.currentValue.trim().length > 0;
+        let choices = this.formatChoices(this.props.choices);
+
+        if (showCreateButton) {
+            choices = [
+                {
+                    text: `${this.state.currentValue}_CREATE_`,
```

Comment:
It's used to differentiate between the create button and the actual choices. This way when the user clicks on the create button, calling `existsInChoices` returns false and we know it's the create button.
It's also prefixed with the current input text so that `InputChip`'s `AutoComplete` always shows it.